core/services/mongodb-remote/src/main/java/com/mongodb/stitch/core/services/mongodb/remote/ChangeStream.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2018-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.stitch.core.services.mongodb.remote;
+
+import com.mongodb.stitch.core.internal.net.StitchEvent;
+import com.mongodb.stitch.core.internal.net.Stream;
+
+import java.io.Closeable;
+import java.io.IOException;
+
+import org.bson.BsonValue;
+
+/**
+ * User-level abstraction for a stream returning {@link ChangeEvent}s from the server.
+ *
+ * @param <UserT> The type returned to users of this API when the next event is requested.
```

Comment:
[naming] I think EventT might be a better name for this. UserT makes it sound like it's a type representing a user.

Reply:
done